agent/provider/tests/manifest_negotiator.rs

```diff
@@ -121,11 +105,35 @@ fn manifest_negotiator_test_accepted_because_of_no_payload() {
     Some("Didn't match any Rules"); // error msg
     "Rejected because domain NOT whitelisted"
 )]
+#[test_case(
+    r#"{"outbound": {"enabled": true, "everyone": "whitelist", "audited-payload": {"default": {"mode": "none", "description": ""}}}}"#, // rulestore config
+    r#"["https://hacked-domain.com"]"#,
+    Some("Didn't match any Rules"); // error msg
+    "Rejected because domain NOT whitelisted - domain ending is the same"
+)]
 #[test_case(
     r#"{"outbound": {"enabled": true, "everyone": "whitelist", "audited-payload": {"default": {"mode": "none", "description": ""}}}}"#, // rulestore config
     r#"["https://domain.com"]"#,
     None; // error msg
-    "Accepted because everyone whitelist matched"
+    "Accepted because everyone whitelist matched - strict matching"
+)]
+#[test_case(
+    r#"{"outbound": {"enabled": true, "everyone": "whitelist", "audited-payload": {"default": {"mode": "none", "description": ""}}}}"#, // rulestore config
+    r#"["https://hacked-regex.com.mydomain.com"]"#,
+    None; // error msg
+    "Accepted because everyone whitelist matched - regex matching"
+)]
+#[test_case(
+    r#"{"outbound": {"enabled": true, "everyone": "whitelist", "audited-payload": {"default": {"mode": "none", "description": ""}}}}"#, // rulestore config
+    r#"["https://hacked-well-formed-regex.com.mydomain.com"]"#,
+    None; // error msg
+    "Rejected because everyone whitelist NOT matched - regex matching"
+)]
+#[test_case(
+    r#"{"outbound": {"enabled": true, "everyone": "whitelist", "audited-payload": {"default": {"mode": "none", "description": ""}}}}"#, // rulestore config
+    r#"["https://valid.well-formed-regex.com"]"#,
+    None; // error msg
+    "Accepted because everyone whitelist matched - regex matching subdomain"
```

Comment:
Whitelist-specific cases should be extended in `manifest_negotiator_test_whitelist` test (as it is created to test all whitelist cases), I would keep `manifest_negotiator_test_manifest_with_urls` focused on negotiations logic with simple whitelisting, and not complicated cases.

Reply:
I noticed that the same tests are duplicated multiple times (for example everyone -> whitelist, audited payload -> none is verified for matching and not matching multiple times).
I added these here because there was already a whitelist did not match test in this bunch, did not want to review the whole test file as part of this PR.